packages/remix/src/infrastructure/logto/create-client.ts

```diff
@@ -0,0 +1,16 @@
+import LogtoClient, { LogtoConfig } from '@logto/node';
+
+import { LogtoStorage } from './create-storage';
+
+export const makeLogtoClient =
+  (config: LogtoConfig, storage: LogtoStorage) =>
+  // Have to deactivate the eslint rule here as the `LogtoClient`
+  // awaits a `navigate` function.
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  (navigate: (url: string) => void = () => {}) => {
+    return new LogtoClient(config, { storage, navigate });
+  };
```

Comment:
I'm just wondering how Remix deals with navigation?
For example in our browser SDK, we use `window.location.href`, and in Next.js SDK we use `response.redirect(url)`.

Reply:
It handles it similar to how Next.js handles redirects. You can find an example in this controller [HandleSignInCallbackController.ts#L51](https://github.com/openformation/js/blob/remix-logto/packages/remix/src/useCases/handleSignInCallback/HandleSignInCallbackController.ts#L51)

Reply:
Well, it doesn't look like what I original suggested. I was trying to make this `create-client.ts` a class instead of the two handlers. But the current design will also do the trick.
So I'm good with it for now. If we come up with a better idea in the future, we'll definitely update and merge back to your repo.